dfsclient/src/main/java/org/hpccsystems/dfs/client/FileUtility.java

```diff
@@ -195,18 +199,206 @@ public JSONArray generateResultsMessage()
         PARQUET
     };
 
-    /**
-     * Returns command line options for the read operation.
-     * @return Options
-     */
-    public static Options getReadOptions()
+    private static class SplitEntry
+    {
+        public long recordCount = 0;
+        public long splitStart = 0;
+        public long splitEnd = 0;
+
+        public JSONObject toJson()
+        {
+            JSONObject res = new JSONObject();
+            res.put("recordCount", recordCount);
+            res.put("splitStart", splitStart);
+            res.put("splitEnd", splitEnd);
+
+            return res;
+        }
+
+        public static SplitEntry fromJson(JSONObject json)
+        {
+            SplitEntry split = new SplitEntry();
+            split.recordCount = json.getLong("recordCount");
+            split.splitStart = json.getLong("splitStart");
+            split.splitEnd = json.getLong("splitEnd");
+
+            return split;
+        }
+    }
+
+    private static class SplitTable
+    {
+        public List<SplitEntry> splits = new ArrayList<SplitEntry>();
+        private long splitStride = 1;
+        private int maxSplitEntries = 100;

```

Comment:
It is pretty arbitrary, but I did change this to be a power of two to make the splits more efficient. It might make sense in the future to have this be user configurable, but requiring a file part to be split by more than 2 orders of magnitude is unlikely.